src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -240,6 +243,7 @@ public void trimOldSnapshots() throws Exception {
     trimMap(shardRqMetricsMap, RQ_SNAPSHOTS);
     trimMap(httpRqMetricsMap, HTTP_RQ_SNAPSHOTS);
     trimMap(masterEventMetricsMap, MASTER_EVENT_SNAPSHOTS);
+	trimMap(faultDetectionMetricsMap, FAULT_DETECTION_SNAPSHOTS);
```

Comment:
Do we need to add a new snapshot type or we can re-use the MASTER_EVENT_SNAPSHOTS here ?

Reply:
We need a new one. As far as i understand, snapshot contains table schema. Master events and Fault Detection events though both are being used to calculate latency and failure metrics have quite different schemas.